cocos/core/game.ts

```diff
@@ -270,6 +139,19 @@ export class Game extends EventTarget {
      */
     public static readonly EVENT_RENDERER_INITED: string = 'renderer_inited';
 
+    public static readonly EVENT_PRE_SETTINGS_INIT = 'pre_settings_init';
```

Comment:
initialization process callback to allow user or plugin or other system inject some custom logic